test/reline/test_key_actor_emacs.rb

```diff
@@ -1416,6 +1416,17 @@ def test_incremental_search_history_saves_and_restores_last_input
     assert_line_around_cursor('abcd', '')
   end
 
+  def test_end_of_history
+    Reline::HISTORY.concat(['abc', '123'])
+    input_keys("\C-rab\C-jd")
+    assert_line_around_cursor('d', 'abc')
+    # \M->: clear input, move history to end, keep edited line as previous history
+    input_key_by_symbol(:end_of_history)
+    assert_line_around_cursor('', '')
+    input_keys("\C-p")
+    assert_line_around_cursor('dabc', '')
```

Comment:
It should be `'123', ''`
I tested with `irb --readline` (`Readline::VERSION #=> 8.2`)